### DISCUSSION TITLE
Deep Nested Mutations

### DISCUSSION THREAD

**Discussion by hadiranjbar25** [Netflix/dgs-framework#598] [Q&A]
Hi
My mutation schema is like this

```
type A {
    create(input: Input!): Response!
    update(input: Input!): Response!
}

type Mutation {
    myMutation1: A
    myMutation2: B
}
```

How should my datafetcher for create look like? 

```
  @DgsData(parentType = DgsConstants.MUTATION.TYPE_NAME, field = DgsConstants.MUTATION.Create)
  public Response createA(@InputArgument("input") final Input input) {
    return new Response();
  }
```
I also have problem with **dgsQueryExecutor**  in my tests. I have no idea how to create query and projection for this schema

**Comment by paulbakker** [comment]
Maybe I'm missing the point, but your schema looks a little weird. In your schema the `return` type is `A`/`B`, while it looks like `A` is the definition of the mutation itself.

Shouldn't your schema look like the following? Of course there are multiple ways to model this.

```
type Mutation {
   createA(input: Input!): Response!
   updateA(input: Input!): Response!

   createB(input: Input!): Response!
   updateB(input: Input!): Response!
}
```

Anyway, based on my schema above, you would have a datafetcher like this:

```java
@DgsMutation
public Response createA(@InputArgument final Input input) {

}
```

**Reply by hadiranjbar25** [reply]
I tried this. But, I get this error
`com.netflix.graphql.dgs.exceptions.QueryException: The field at path '/myMutation1/create' was declared as a non null type, but the code involved in retrieving data has wrongly returned a null value.  The graphql specification requires that the parent field be set to null, or if that is non nullable that it bubble up null to its parent and so on. The non-nullable type is 'Response' within parent type 'Response'
`
Our schema for mutation is nested. That's why we have a multilevel mutation. Dgs does not generate queryGenerator for the child field. It only generates it for myMutation1 and 2

**Reply by hadiranjbar25** [reply]
Does Dgs has a support for nested queries? Our use case is something like this:
https://graphql-api.com/guides/schema/using-nested-mutations/
